### ISSUE TITLE
url check at bot start

### ISSUE THREAD

**Issue by DieNastie** [ilpersi/BHBot#156]
this might be just one line of code
let the bot check if the url at the start of chromium is already the correct one

i use `useDoNotShareURL 1` as my homepage but the bot then frist goes to kong only then to load the actual url after that
thats 3 siteloads back to back

**Comment by ilpersi** [comment]
I would like to have it so as well. 😄

**TL;DR**
I'll try to keep this simple and the reasons why this is the way you see now are that:
- the URL "without the Kong interface"is different for every user (the base web address is the same, the parameters  used with it are different for each user)
- once you get the URL for the first time, it is changing over time (so what works today may not work in the future).

To handle this, the bot has to:
- log in as you would normally do
- retrieve the correct URL
- restart the session using the new URL

The restart is required cause when the bot is not using the kong interface the graphics are in a different position and many different considerations have to be done at the code level to handle this.

I tweaked the logic many times before getting to the current solution that is the most conservative one: the URL switch is done one single time per run so if you leave the bot running for days this should not bother you.

The current solution also accounts for the safest security posture: if for any reason the URL is leaked to someone that is not you, this person will be able to use your account just pasting the URL in a browser (no user & pass would be required to log in) that's why the URL is not saved anywhere and it is only available in memory while the bot is running.

Most of the explanations above are also the reason why this parameter is "experimental" and not present in the standard settings.ini.

I hope this is clear enough. 👍 

If you look in Chrome history you can see how the URL is changing over time:

![image](https://user-images.githubusercontent.com/43588107/79077893-dc71ad00-7d04-11ea-8cda-e72af8155793.png)

**Comment by DieNastie** [comment]
ok this is all well but would at least the frist check be too much to ask?
that would at least get rid of the frist kong load

**Comment by ilpersi** [comment]
What the check should be?

**Comment by DieNastie** [comment]
just to check for 
https://www.kongregate.com/games/Juppiomenz/bit-heroes
its simple but its a start

**Comment by ilpersi** [comment]
I don't get it.

I check for https://www.kongregate.com/games/Juppiomenz/bit-heroes .... And?

**Comment by DieNastie** [comment]
if that url is already being loaded skip loading it

**Comment by ilpersi** [comment]
Maybe I was not clear in my first explanation, so I will try to put in more simple terms.

When the bot is started the DO_NOT_SHARE_URL **is unknwon**. The only way to retrieve it is to load the standard URL (https://www.kongregate.com/games/Juppiomenz/bit-heroes) once.

There's no way that loading https://www.kongregate.com/games/Juppiomenz/bit-heroes can be skipped.

**Comment by DieNastie** [comment]
no you were clear enough

issue is the normal kong link is being called twice
chrome starts and loads BH on kong cus that is the homepage
then the bots comes and loads the same url again
after that it loads the DO_NOT_SHARE_URL

with my slow connection thats a 5 sec loss (or more) every restart/crash

**Comment by ilpersi** [comment]
I think I get it now!

You have the standard URL as homepage and you want the bot to check for that.

I'll have a look 🙂👍🏻

**Comment by ilpersi** [comment]
v45 is in with this feature packed in
